### PR TITLE
Chores/cleanup snake case camel case fields #159669320

### PR DESCRIPTION
**Pull request by fddayan**
https://www.pivotaltracker.com/story/show/159669320

### CODE REVIEW

**Comment by chrisdolendo** [comment]
@fddayan We have some fields still unmapped -- ADA Priorities do not save (ReservedPrioritySection.js) and are still camelcase-- can you have a look at all the paper application components to make sure we caught all of them?

**Review by chrisdolendo — CHANGES_REQUESTED** [review]
Still missing some fields 😬

File: app/javascript/components/applications/application_form/ReservedPrioritySection.js [hunk 1]
```diff
@@ -27,7 +26,7 @@ const ReservedPrioritySection = ({editValues, listing}) => {
         <div className="small-6 columns">
           <label>Developmentally Disabled</label>
           <p className="form-note margin-bottom">One or more units is reserved for applicants who are developmentally disabled. Select "Yes" below if a household member is developmentally disabled.</p>
-          <Select field="hasDevelopmentalDisability" options={yes_no_options} />
+          <Select field="has_developmentaldisability" options={yes_no_options} />
```
Comment (akegan) [inline app/javascript/components/applications/application_form/ReservedPrioritySection.js]: missing a _ here between developmental and disability

File: app/javascript/components/applications/application_form/utils.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    get(p, 'application_member.date_of_birth'),
+  ], ',')
+
+export const naturalKeyFromMember = (member) => {
+  return `${member.first_name},${member.last_name},${member.date_of_birth}`
+}
+
+export const FIELD_NAME = 'preferences'
+
+export const buildFieldId = (i, field) => `${FIELD_NAME}.${i}.${field}`
```
Comment (akegan) [inline app/javascript/components/applications/application_form/utils.js]: didn't build field id already exist? why is this here again?

File: app/javascript/components/applications/application_form/utils.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    get(p, 'application_member.first_name'),
+    get(p, 'application_member.last_name'),
+    get(p, 'application_member.date_of_birth'),
+  ], ',')
+
+export const naturalKeyFromMember = (member) => {
+  return `${member.first_name},${member.last_name},${member.date_of_birth}`
+}
+
+export const FIELD_NAME = 'preferences'
```
Comment (akegan) [inline app/javascript/components/applications/application_form/utils.js]: what's this?

**Review by akegan — CHANGES_REQUESTED** [review]

File: app/javascript/components/applications/application_form/utils.js (hunk 3 above)
Comment (fddayan) [reply app/javascript/components/applications/application_form/utils.js]: good catch, I think I had this file file open from a different branch and I might have hit `save all` and it was re saved in this branch. I remove this file the one under preferences is the right one

File: app/javascript/components/applications/application_form/ReservedPrioritySection.js (hunk 1 above)
Comment (fddayan) [reply app/javascript/components/applications/application_form/ReservedPrioritySection.js]: fixed

**Comment by fddayan** [comment]
@akegan ready

**Review by akegan — APPROVED** [review]

**Review by chrisdolendo — APPROVED** [review]

**Comment by chrisdolendo** [comment]
@fddayan I tested this and did a code check and all looks good. Merging now.
